### PR TITLE
fix: GraphQL rehaul

### PR DESCRIPTION
**Pull request by spark-c**
Rehaul of [GraphQL datasource](https://docs.appsmith.com/reference/datasources/graphql) docs page

### CODE REVIEW

File: website/docs/reference/datasources/graphql.md [hunk 1]
```diff
@@ -3,496 +3,497 @@ sidebar_position: 8
...
-## Create GraphQL API
-
-To add a GraphQL API, navigate to **Explorer** >> Click on the **+** icon (next to Datasources) >> Select **Authenticated GraphQL API** under APIs.
-
-:::info
-You can choose to create a one-off query by selecting [_**Create New GraphQL API**_](graphql.md#datasource-page)_**.**_
-:::
-
-<VideoEmbed host="youtube" videoId="lIb2SGnQFm4" title="Create GraphQL Datasource" caption="Create GraphQL Datasource"/>
+## Connect to GraphQL API
```
Comment (danciaclara) [inline website/docs/reference/datasources/graphql.md:10]:
```suggestion
## Connect to GraphQL
```

File: website/docs/reference/datasources/graphql.md [hunk 2]
```diff
@@ -3,496 +3,497 @@ sidebar_position: 8
...
-![Create GraphQL Datasource](</img/Screenshot_2022-09-07_at_11.43.59_AM.png>)<figure><figcaption align="center"><i>Create GraphQL Datasource</i></figcaption></figure>
+<figure>
+  <img src="/img/graphql-datasource-config.png" style={{width: "100%", height: "auto"}} alt="Configuring a GraphQL datasource" />
+  <figcaption align="center"><i>Configuring a GraphQL datasource.</i></figcaption>
+</figure>
 
-:::info
-All required fields are suffixed with an asterisk.
+:::tip
+If you want to connect to a local database, you can use a service like ngrok to expose it. For more information, see [How to connect to local database on Appsmith](/advanced-concepts/more/how-to-work-with-local-apis-on-appsmith).
```
Comment (danciaclara) [inline website/docs/reference/datasources/graphql.md:22]:
```suggestion
If you want to connect to a local API, you can use a service like ngrok to expose it. For more information, see [How to connect to local database on Appsmith](/advanced-concepts/more/how-to-work-with-local-apis-on-appsmith).
```

File: website/docs/reference/datasources/graphql.md [hunk 3]
```diff
@@ -3,496 +3,497 @@ sidebar_position: 8
...
 :::
 
-#### URL
-
-Use this field to add the API URL you want to access. For instance, if you want to visit [GitHub GraphQL API](https://docs.github.com/en/graphql), enter [https://api.github.com/graphql ](https://api.github.com/graphql)in the URL field.
+To connect to your database, Appsmith needs the following parameters. All required fields are suffixed with an asterisk (*).
```
Comment (danciaclara) [inline website/docs/reference/datasources/graphql.md:25]:
```suggestion
To connect to a GraphQL API endpoint, Appsmith needs the following parameters. All required fields are suffixed with an asterisk (*).
```

File: website/docs/reference/datasources/graphql.md [hunk 4]
```diff
@@ -3,496 +3,497 @@ sidebar_position: 8
...
-2. Request Authorization;
-3. Response Caching;
-4. Response Cookies.
+**Query Parameters:** provide any parameters that should be passed as key-value pairs in the URL of your requests.
 
-#### Query Parameters
+**Send Appsmith signature header:** choose whether to include a special token in your request headers to help prove authenticity and integrity. For more information, read [signature headers](/core-concepts/connecting-to-data-sources/authentication/signature-header-in-api-actions).
 
-A parameter is a piece of information you provide to a query during its execution. In the query, parameters can be used independently or as a component of a larger expression to build a criterion. The API you integrate allows you to include one or more[ query parameters](/core-concepts/data-access-and-binding/querying-a-database) as part of the request.
+**Authentication Type:** choose the style of authentication to use for your queries.
```
Comment (danciaclara) [inline website/docs/reference/datasources/graphql.md:35]: we also have a page for authentication type, so why not redirect there for more info like the others

File: website/docs/reference/datasources/graphql.md [hunk 5]
```diff
@@ -3,496 +3,497 @@ sidebar_position: 8
...
-Self-signed certificates are sometimes used in testing or internal environments, as they provide incoming/outgoing data encryption and can be created at no cost.
-
-The GraphQL Datasource allows you to upload your own self-signed certificate within the API's configuration page. You can follow this [document ](/core-concepts/connecting-to-data-sources/authentication/self-signed-certificates)to learn more about Self-signed certificates.
-
-:::info
-This information needs to be provided in .PEM (Privacy Enhanced Mail) format, as shown [here](/core-concepts/connecting-to-data-sources/authentication/self-signed-certificates).
-:::
+<figure>
+  <img src="/img/graphql-query-config.png" style={{width: "100%", height: "auto"}} alt="Configuring a GraphQL query." />
+  <figcaption align="center"><i>Configuring a GraphQL query.</i></figcaption>
```
Comment (danciaclara) [inline website/docs/reference/datasources/graphql.md:43]:
```suggestion
  <figcaption align="center"><i>Creating a GraphQL query</i></figcaption>
```

File: website/docs/reference/datasources/graphql.md [hunk 6]
```diff
@@ -3,496 +3,497 @@ sidebar_position: 8
...
+</figure>
 
-## Create Queries
+You can write [queries](https://docs.appsmith.com/core-concepts/data-access-and-binding/querying-a-database/query-settings) to fetch or write data by selecting the **+ New Query**  button on the Authenticated GraphQL datasource page, or by clicking (**+**) next to **Queries/JS** in the **Explorer** tab and selecting your GraphQL datasource. You'll be brought to a new query screen where you can write queries.
 
-You can add queries to the newly created GraphQL datasource in one of the below ways:
+GraphQL queries are written in the "Body" tab of the query screen. Use the "Query" window to construct your query or mutation, and the adjacent "Query Variables" window to add any variables to map into your query.
```
Comment (danciaclara) [inline website/docs/reference/datasources/graphql.md:48]: let all UI element be in bold text. for eg: **Body** tab

File: website/docs/reference/datasources/graphql.md [hunk 7]
```diff
@@ -3,496 +3,497 @@ sidebar_position: 8
...
 
-#### GraphQL Datasource Page
+### Fetch records
```
Comment (danciaclara) [inline website/docs/reference/datasources/graphql.md:50]: why are not using the same public API used in the current docs?

**Review by danciaclara — CHANGES_REQUESTED** [review]

File: website/docs/reference/datasources/graphql.md (hunk 4 above)
Comment (spark-c) [reply website/docs/reference/datasources/graphql.md:35]: Added the link -- it seems like the content on these pages (particularly the "Basic" one) isn't super helpful, but we'll probably rehaul those pages in the future to be more valuable

**Review by danciaclara — DISMISSED** [review]

**Review by danciaclara — APPROVED** [review]

**Pull request merged**